src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule73wherejavadocrequired/InputMissingJavadocTypeCorrect.java

```diff
@@ -53,4 +52,112 @@ private interface MyInterfacePrivate {}
   public void myMethod() {
     class MyMethodClass {}
   }
+
+  /** some javadoc. */
+  protected int testProtectedMethod1() {
+    return 0;
+  }
+
+  /** some javadoc. */
+  protected void testProtectedMethod2() {}
+
+  /** some javadoc. */
+  public void testPublicMethod() {}
+
+  private void testPrivateMethod() {}
+
+  void testPackagePrivateMethod() {}
+
+  /** some javadoc. */
+  protected InputMissingJavadocTypeCorrect() {}
+
+  /** some javadoc. */
+  public InputMissingJavadocTypeCorrect(String arg) {}
+
+  private InputMissingJavadocTypeCorrect(int arg) {}
+
+  InputMissingJavadocTypeCorrect(double arg) {}
+
+  /** some javadoc. */
+  protected @interface ProtectedAnnotation {}
+
+  /** some javadoc. */
+  public @interface PublicAnnotation {}
+
+  private @interface PrivateAnnotation {}
+
+  @interface PackagePrivateAnnotation {}
+
+  /** some javadoc. */
+  protected class InnerProtected {
+    protected void testProtectedInnerMethod() {}
+
+    public void testPublicInnerMethod() {}
+
+    private void testPrivateInnerMethod() {}
+
+    void testPackagePrivateInnerMethod() {}
+
+    protected InnerProtected() {}
+
+    InnerProtected(String arg) {}
+
+    private InnerProtected(double arg) {}
+  }
+
+  /** some javadoc. */
+  public enum MyEnumPublic2 {
+    TEST;
+
+    /** some javadoc. */
+    public void testPublicEnumMethod() {}
+
+    /** some javadoc. */
+    protected void testProtectedEnumMethod() {}
+
+    private void testPrivateEnumMethod() {}
+
+    void testPackagePrivateEnumMethod() {}
+
+    private MyEnumPublic2() {}
+
+    MyEnumPublic2(String arg) {}
+  }
+
+  /** some javadoc. */
+  protected enum MyEnumProtected {
+    TEST;
+
+    public void testPublicEnumMethod() {}
+
+    protected void testProtectedEnumMethod() {}
+
+    private void testPrivateEnumMethod() {}
+
+    void testPackagePrivateEnumMethod() {}
+
+    private MyEnumProtected() {}
+
+    MyEnumProtected(String arg) {}
+  }
+
+  /** some javadoc. */
+  protected interface MyInterfaceProtected {
+    public void testPublicInterfaceMethod();
+
+    private void testPrivateInterfaceMethod() {}
+
+    void testPackagePrivateInterfaceMethod();
+  }
+
+  /** some javadoc. */
+  public interface MyInterfacePublic2 {
+    /** some javadoc. */
+    public void testPublicInterfaceMethod();
+
+    private void testPrivateInterfaceMethod() {}
+
+    /** some javadoc. */
+    void testPackagePrivateInterfaceMethod();
```

Comment:
this is false positive, if a method isn't public or protected then it does not require javadoc but we're giving warning for it though when I try to run the same xpath query for same code locally like this:

```
$ cat Test.java
public class Test {
  public interface MyInterfacePublic {
    public void testPublicInterfaceMethod();

    private void testPrivateInterfaceMethod() {}

    void testPackagePrivateInterfaceMethod();
  }
}

$ java -jar checkstyle-10.18.1-all.jar -b "//*[self::METHOD_DEF or self::CTOR_DEF or self::ANNOTATION_FIELD_DEF or self::COMPACT_CTOR_DEF][ancestor::*[self::INTERFACE_DEF or self::CLASS_DEF or self::RECORD_DEF or self::ENUM_DEF][not(./MODIFIERS/LITERAL_PUBLIC)]]" Test.java 
```

there's no output for it.